megatron/utils.py

```diff
@@ -234,6 +234,9 @@ def print_rank_0(message):
     else:
         print(message, flush=True)
 
+def print_flush(message):
+    print(f"[{torch.distributed.get_rank()}] {message}", flush=True)
```

Comment:
@zhenghh04 is it necessary to print from every rank here?

In my tests, I've changed this to:

```python
import ezpz as ez
import logging

RANK = ez.get_rank()
log = logging.getLogger(__name__)
log.setLevel("INFO") if RANK == 0 else log.setLevel("CRITICAL")

def print_flush(message):
    log.info(f'[{RANK}]: {message}')
```


which seems to have been working OK for me so far

Reply:
@saforem2 This is for the place where the work was done purely by rank 0 in the original code. Now since we distribute the work among all the ranks, so I made all the ranks print the info which will be equivalent to the original code.